scripts/release/create-release-metadata-for-ghcup.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
```

Comment:
```suggestion
#!/usr/bin/env bash
```

Reply:
Sure.

Also wanted to verify if we'd need to `cd` into a directory for artifacts/downloads here -> https://github.com/arjunkathuria/cabal/blob/release-metadata-script/scripts/release/create-release-metadata-for-ghcup.sh#L12-L13

Reply:
Done.

Reply:
> ```suggestion
> #!/usr/bin/env bash
> ```

Why? This isn't more portable. 

https://stackoverflow.com/a/53116875

Reply:
This link talks about the Bourne Shell (`sh`) but this PR uses `bash`. On my FreeBSD machine `bash` is located elsewhere:

```
% whereis bash
bash: /usr/local/bin/bash /usr/local/man/man1/bash.1.gz
```

Reply:
I don't mind this script being Linux-only, but then in this case there should be a test on `uname` at the beginning of the script.

Reply:
Not every Linux has `/bin/bash` either. For example, I use NixOS, for which only the proposed change will work. I assume, Guix is the same. I highly recommend to apply the change with /usr/bin/env: from what I heard in the NixOS community, this is *much* more portable.

Reply:
Oh, it was actually applied. Good.

Reply:
> from what I heard in the NixOS community, this is much more portable.

Yeah, they've also been spreading misinformation that this is recommended by posix standard :p 

As per posix: both are equally non-portable.

I believe switching to `/bin/sh` is practically more portable. Since NixOS isn't properly FHS compatible, I consider it an outlier.

Reply:
if its agreed up, i can change the other script (the download script) to use `/usr/bin/env bash` as well

Reply:
Posix or not, this practically covers more systems.

Reply:
Practically `#!/bin/sh` is the most portable option. A system might have no `bash` at all. Any particular reason not to use plain old `sh`?